mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -517,69 +516,75 @@ class WidgetAdapter(
         private val connection: Connection,
         colorMapper: ColorMapper
     ) : LabeledItemBaseViewHolder(inflater, parent, R.layout.widgetlist_slideritem, connection, colorMapper),
-        SeekBar.OnSeekBarChangeListener {
-        private val seekBar: SeekBar = itemView.findViewById(R.id.seekbar)
+        Slider.OnSliderTouchListener, Slider.LabelFormatter {
+        private val seekBar: Slider = itemView.findViewById(R.id.seekbar)
         private var boundWidget: Widget? = null
 
         init {
-            seekBar.setOnSeekBarChangeListener(this)
-            val now = Calendar.getInstance()
-            if (now.get(Calendar.DAY_OF_MONTH) == 31 && now.get(Calendar.MONTH) == Calendar.OCTOBER) {
-                seekBar.thumb = ContextCompat.getDrawable(itemView.context, R.drawable.ic_halloween_orange_24dp)
-            }
+            seekBar.addOnSliderTouchListener(this)
+            seekBar.setLabelFormatter(this)
         }
 
         override fun bind(widget: Widget) {
             super.bind(widget)
             boundWidget = widget
 
-            val stepCount = (widget.maxValue - widget.minValue) / widget.step
-            seekBar.max = Math.ceil(stepCount.toDouble()).toInt()
-            seekBar.progress = 0
+            // Fix "The stepSize must be 0, or a factor of the valueFrom-valueTo range" exception
+            seekBar.valueTo = widget.maxValue - (widget.maxValue - widget.minValue).rem(widget.step)
+            seekBar.valueFrom = widget.minValue
+            seekBar.stepSize = if (widget.step == 1F) 0F else widget.step
```

Comment:
Why special case 1 here?

Reply:
If `stepSize > 0` the slider is set to discrete. The step size sent by the server is 1 if no step size is defined.
IMO if the user defines no step size the slider shouldn't be discrete. Checking for `step == 1` here is not the perfect solution, though, as it may be requested by the user.

Reply:
Thinking about this, the better solution would be to just set the step size for now and see what feedback we get in https://github.com/material-components/material-components-android/issues/1233.